core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -617,12 +594,12 @@ public IndexMetaDataDiff(StreamInput in) throws IOException {
                 new DiffableUtils.DiffableValueSerializer<String, Custom>() {
                     @Override
                     public Custom read(StreamInput in, String key) throws IOException {
-                        return lookupPrototypeSafe(key).readFrom(in);
+                        return registry.getIndexMetadataPrototype(key).readFrom(in);
```

Comment:
Use `getIndexMetadataPrototypeSafe` so you throw a nice exception instead of NPE?